src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Unix.cs

```diff
@@ -116,31 +115,87 @@ public override void Unpin()
+                // in the buffer. Here the baseAddress points to the first valid allocation and poisonPageAddress points to the first invalid location.
+                // For `PoisonPagePlacement.Before` the first page is made inaccessible using mprotect and baseAddress points to the start of the second page.
+                // The allocation and protection is at the granularity of a page. Thus, `PoisonPagePlacement.Before` configuration has an additional accessible
+                // memory at the end of the page (bytes equivalent to `pageSize - (byteLength % pageSize)`).
+                // For `PoisonPagePlacement.After`, we adjust the baseAddress so that inaccessible memory is at the `byteLength` offset from the baseAddress.
+                IntPtr baseAddress = buffer + pageSize;
+                IntPtr poisonPageAddress = buffer;
+                if (placement == PoisonPagePlacement.After)
+                {
+                    baseAddress = buffer + (allocationSize - pageSize - byteLength);
+                    poisonPageAddress = buffer + (allocationSize - pageSize);
+                }
+
+                // Protect the page before/after based on the poison page placement.
+                if (MProtect(poisonPageAddress, (ulong)pageSize, (int)MemoryMappedProtections.PROT_NONE) == -1)
+                {
+                    throw new InvalidOperationException($"Failed to mark page as a poison page using mprotect with error :{Marshal.GetLastPInvokeError()}.");
+                }
+
+                MMapHandle retVal = new MMapHandle(buffer, (ulong)allocationSize);
+                retVal.SetHandle(baseAddress); // this base address would be used as the start of Span that is used during unit testing.
                 return retVal;
             }
 
-            // Do not provide a finalizer - SafeHandle's critical finalizer will
-            // call ReleaseHandle for you.
-
             public override bool IsInvalid => (handle == IntPtr.Zero);
 
             protected override bool ReleaseHandle()
             {
-                Marshal.FreeHGlobal(handle);
+                MUnmap(buffer, allocationSize);
                 return true;
             }
+
+            [Flags]
+            private enum MemoryMappedProtections
+            {
+                PROT_NONE = 0x0,
+                PROT_READ = 0x1,
+                PROT_WRITE = 0x2,
+                PROT_EXEC = 0x4
+            }
+
+            [Flags]
+            private enum MemoryMappedFlags
+            {
+                MAP_SHARED = 0x01,
+                MAP_PRIVATE = 0x02,
+                MAP_ANONYMOUS = 0x10,
+            }
+
+            private const string SystemNative = "libSystem.Native";
+
+            // NOTE: Shim returns null pointer on failure, not non-null MAP_FAILED sentinel.
+            [DllImport(SystemNative, EntryPoint = "SystemNative_MMap", SetLastError = true)]
```

Comment:
nit: you should be able to access `Interop.Libraries.SystemNative` here for the lib name constant

Reply:
or maybe just consume Interop.MMap.cs which is already in the shared location?

Reply:
Because we directly reference `BoundedMemory.cs` and its implementations in `Sve_r.csproj` and `Sve_ro.csproj`, I do not have access to any other API in the `TestUtilities.csproj` nor do I have access to internals; which means I cannot access `Interop.Libraries.SystemNative`.